File: web/src/services/task.ts
import request from '../utils/request'
import type { PageResult } from './audit'

export interface TaskItem {
  id: number
  name: string
  type: string
  command: string
  status: string
  created_by: number
  created_by_name: string
  total_count: number
  success_count: number
  fail_count: number
  started_at: string
  finished_at: string
  created_at: string
}

export interface TaskResultItem {
  id: number
  task_id: number
  asset_id: number
  hostname: string
  ip: string
  status: string
  stdout: string
  stderr: string
  exit_code: number
  duration: number
}

export interface ExecuteTaskParams {
  name: string
  command: string
  asset_ids: number[]
}

export function executeTask(data: ExecuteTaskParams): Promise<TaskItem> {
  return request.post('/tasks/execute', data)
}

export function getTasks(params?: { status?: string; page?: number; page_size?: number }): Promise<PageResult<TaskItem>> {
  return request.get('/tasks', { params })
}

export function getTask(id: number): Promise<TaskItem> {
  return request.get(`/tasks/${id}`)
}

export function getTaskResults(id: number): Promise<TaskResultItem[]> {
  return request.get(`/tasks/${id}/results`)
}

File: internal/dto/task.go
package dto

// TaskExecuteRequest 批量执行请求
type TaskExecuteRequest struct {
	Name     string `json:"name" binding:"required,max=128"`
	Command  string `json:"command" binding:"required"`
	AssetIDs []uint `json:"asset_ids" binding:"required"`
}

// TaskListRequest 任务列表请求
type TaskListRequest struct {
	Status string `form:"status"`
	Page   int    `form:"page"`
	Size   int    `form:"page_size"`
}

// TaskResponse 任务响应
type TaskResponse struct {
	ID            uint   `json:"id"`
	Name          string `json:"name"`
	Type          string `json:"type"`
	Command       string `json:"command"`
	Status        string `json:"status"`
	CreatedBy     uint   `json:"created_by"`
	CreatedByName string `json:"created_by_name"`
	TotalCount    int    `json:"total_count"`
	SuccessCount  int    `json:"success_count"`
	FailCount     int    `json:"fail_count"`
	StartedAt     string `json:"started_at"`
	FinishedAt    string `json:"finished_at"`
	CreatedAt     string `json:"created_at"`
}

// TaskResultResponse 任务结果响应
type TaskResultResponse struct {
	ID       uint   `json:"id"`
	TaskID   uint   `json:"task_id"`
	AssetID  uint   `json:"asset_id"`
	Hostname string `json:"hostname"`
	IP       string `json:"ip"`
	Status   string `json:"status"`
	Stdout   string `json:"stdout"`
	Stderr   string `json:"stderr"`
	ExitCode int    `json:"exit_code"`
	Duration int64  `json:"duration"`
}
